keps/0009-operator-toolkit.md

```diff
@@ -106,93 +100,80 @@ An operator package is a folder that contains all of the manifests needed to cre
 
 #### operator.yaml
 
-`operator.yaml` is the base definition of an operator. It follows the following format, extracted from the MySQL example:
+`operator.yaml` is the base definition of an operator.  It follows the following reference format.
 
 ```yaml
-name: operator
-description: operator
-version: "5.7"
-kudoVersion: ">= 0.2.0"
-kubeVersion: ">= 1.14"
+name: str(max=253, lowercase alphanumeric characters, -, and .)
```

Comment:
is this some validation/spec language? if so, should we probably not what we're using here? It kind of looks like a golang to me but the notation says yaml so I am a bit confused...

Reply:
although if this IS some sort of spec language that is already known, I would totally use this in params.yaml so we could get nested values :o

Reply:
it is vaguely close to https://github.com/23andMe/Yamale
the point was to have condensed yet understandable constraints.
even in the psuedo sense... it seems understandable

Reply:
not to replace a real schema at some pt of course

Reply:
I'm torn on this one: if we're talking about defining a real schema then I'd go with [OpenAPI](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#specifying-a-structural-schema) and keep it outside of the KEP. An even better approach would be to generate these to keep them in sync with the rest of the codebase but I don't have a solution right now, so we might need to research our options here first. 

The proposed pseudo schema is neither generated nor is it parsed and raises questions of why it is in this form. In this case, it can simply be examples with YAML comments describing individual fields.